agreementbot/consumer_protocol_handler.go

```diff
@@ -139,9 +139,37 @@ func (w *BaseConsumerProtocolHandler) sendMessage(mt interface{}, pay []byte) er
 		return errors.New(fmt.Sprintf("input message target is %T, expecting exchange.MessageTarget", mt))
 	}
 
+	logMsg := string(pay)
```

Comment:
1) All of this needs to happen inside a check for V(5) logging being on or off
2) Why are you demarshalling as a proposal and as a policy? I dont think we ever send just a policy

Reply:
I did it in case it is ever sent in the future. I suppose it could hinder performance though, so I will change it to just check for a proposal.